File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length()!=t.length())
        {
            return false;
        }

        unordered_map<char, int> sh;
        unordered_map<char, int> th;
        for(int i=0; i<s.length();i++)
        {
            sh[s[i]] += 1;
            th[t[i]] += 1;
        }
        return sh==th;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if(len(s)!=len(t)):
            return False
        sh, th = defaultdict(int),defaultdict(int)
        #countS, countT = {}, {}
        for i in range(len(s)):
            sh[s[i]] +=1; #countS[s[i]] = 1 + countS.get(s[i], 0)
            th[t[i]] +=1;
        return sh==th;

        #python throws key error if normal dicts are use to mimic 
        #cpp behavior use defaultdict